src/components/editor/PropertyComponent.jsx

```diff
@@ -0,0 +1,95 @@
+// Copyright 2019 Stanford University see Apache2.txt for license
+
+import React, { Component } from 'react'
+import InputLiteral from './InputLiteral'
+import InputListLOC from './InputListLOC'
+import InputLookupQA from './InputLookupQA'
+import lookupConfig from '../../../static/spoofedFilesFromServer/fromSinopiaServer/lookupConfig.json'
+import PropTypes from 'prop-types'
+const _ = require('lodash')
+
+export class PropertyComponent extends Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      configuration: getLookupConfigItems(this.props.propertyTemplate)
+    }
+  }
+
+  inputComponentType = (property) => {
+    let config, result
+
+    try {
+      config = this.state.configuration[0].component
+    } catch {
+      // ignore undefined configuration
+    }
+
+    // We do not support mixed list and lookups, so we will just go with the value of the first config item found
+    switch(config) {
```

Comment:
Not blocking but the more get into the specific Sinopia workflows, I'm wondering if the `type` property could reference the React component instead of us needing to this conditional code to determine the specific React component to use.

Reply:
I think we decided that `list` would not be one of the type properties (See https://github.com/LD4P/sinopia/blob/0d0d858343725ba11e4ceec59a6c917e237c37c0/schemas/0.2.0/property-template.json#L71) so we need to check the config details of the lookup source to choose the component. However, this is going to change slightly after QA starts hosting the config (see https://github.com/LD4P/qa_server/issues/51) and will hinge ion whether or not the `action` is a `search`. 

If I am mis-reading your comment let's discuss!

Reply:
Resolving for now, this is a larger conversation as we look towards **M4** support for a [InputLookupSinopia](https://github.com/LD4P/sinopia_editor/issues/442) component to support searching the Sinopia Server's elasticsearch index. Don't think we want to depend on QA to host configurations for non-QA React components (although they are for the `InputListLOC`?) 

There are likely other components coming soon like a [RDFRegistry](https://www.rdaregistry.info/) that QA is not supporting but has already been requested by a cohort member. Another component is a direct Wikidata lookup component that uses the wikidata REST JSON API (I don't think we want to wait if/when QA supports Wikidata when there is already interest in the cohort to use Wikidata directly in Sinopia and a REST JSON api already exists)

Reply:
Point well taken, and we may need to revisit the schema requirements and spefication of different types in property templates